WordPress/Classes/ViewRelated/Post/AztecAttachmentViewController.swift

```diff
@@ -0,0 +1,195 @@
+        navigationItem.leftBarButtonItem = cancelButton
+
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target:self, action: #selector(AztecAttachmentViewController.handleDoneButtonTapped))
+        navigationItem.rightBarButtonItem = doneButton
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.handler.viewModel = self.tableViewModel()
+    }
+
+    // MARK: - Model mapping
+
+    func tableViewModel() -> ImmuTable {
+        let displaySettingsHeader = NSLocalizedString("Web Display Settings", comment: "The title of the option group for editing an image's size, alignment, etc. on the image details screen.")
+
+        let alignmentRow = EditableTextRow(
+            title: NSLocalizedString("Alignment", comment: "Image alignment option title."),
+            value: alignTitles[alignment]!,
+            action: displayAlignmentSelector)
+
+        let sizeRow = EditableTextRow(
+            title: NSLocalizedString("Size", comment: "Image size option title."),
+            value: sizeTitles[size]!,
+            action: displaySizeSelector)
+
+        return ImmuTable(sections: [
+            ImmuTableSection(
+                headerText: displaySettingsHeader,
+                rows: [
+                    alignmentRow,
+                    sizeRow,
+                ],
+                footerText: nil)
+            ])
+    }
+
+
+    // MARK: - Actions
+
+    func displayAlignmentSelector(row: ImmuTableRow) {
+
+        let values = alignTitles.map { (key: TextAttachment.Alignment, value: String) -> TextAttachment.Alignment in
+            return key
+        }
+
+        let titles = values.map { (value: TextAttachment.Alignment) -> String in
+            return alignTitles[value]!
+        }
+
+        let currentValue = alignment
+
+        let dict: [String: Any] = [
+            "DefaultValue": alignment,
+            "Title": NSLocalizedString("Alignment", comment:"Title of the screen for choosing an image's alignment."),
+            "Titles": titles,
+            "Values": values,
+            "CurrentValue": currentValue
+        ]
+
+        guard let vc = SettingsSelectionViewController(dictionary: dict) else {
+            return
+        }
+
+        vc.onItemSelected = { (status: Any) in
+            if let newAlignment = status as? TextAttachment.Alignment {
+                self.alignment = newAlignment
+            }
+            vc.dismiss()
+            self.tableView.reloadData()
+        }
+
+        self.navigationController?.pushViewController(vc, animated: true)
+    }
+
+    func displaySizeSelector(row: ImmuTableRow) {
+        let values = sizeTitles.map { (key: TextAttachment.Size, value: String) -> TextAttachment.Size in
+            return key
+        }
+
+        let titles = values.map { (value: TextAttachment.Size) -> String in
+            return sizeTitles[value]!
+        }
```

Comment:
(Same as above) no need to specify the `value`'s type, nor the return type, since those can be inferred by the compiler